File: typical90/055/55.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
    int n, p, q;
    cin >> n >> p >> q;
    vector<long long> a(n);
    for (int i = 0; i < n; i++) cin >> a[i];
    int ans = 0;
    for (int i = 0; i < n; i++) {
        for (int j = 0; j < i; j++) {
            for (int k = 0; k < j; k++) {
                for (int l = 0; l < k; l++) {
                    for (int m = 0; m < l; m++) {
                        if (a[i] * a[j] % p * a[k] % p * a[l] % p * a[m] % p == q) ans++;
                    }
                }
            }
        }
    }
    cout << ans << '\n';
}

File: typical90/055/55.py
n, p, q = map(int, input().split())
a = list(map(int, input().split()))
ans = 0
for i in range(n):
	for j in range(i):
		for k in range(j):
			for l in range(k):
				for m in range(l):
					if a[i] * a[j] * a[k] * a[l] * a[m] % p == q:
						ans += 1
print(ans)